docs/src/main/asciidoc/artemis-jms-guide.adoc

```diff
@@ -0,0 +1,304 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Using Artemis JMS extension
+include::./attributes.adoc[]
+
+This guide demonstrates how your Quarkus application can use Artemis JMS messaging.
+
+== Prerequisites
+
+To complete this guide, you need:
+
+* less than 15 minutes
+* an IDE
+* JDK 1.8+ installed with `JAVA_HOME` configured appropriately
+* Apache Maven 3.5.3+
+* A running Artemis server, or Docker Compose to start one
+* GraalVM installed if you want to run in native mode.
+
+== Architecture
+
+In this guide, we are going to generate (random) prices in one component.
+These prices are written in an JMS queue (`prices`).
+Another component reads from the `prices` queue and stores the last price.
+The data can be fetch from a browser using a fetch button from a JAX-RS resource.
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step.
+However, you can go right to the completed example.
+
+Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
+
+The solution is located in the `artemis-jms` {quickstarts-tree-url}/artemis-jms[directory].
+
+== Creating the Maven Project
+
+First, we need a new project. Create a new project with the following command:
+
+[source, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=artemis-jms \
+    -Dextensions="artemis-jms"
+----
+
+This command generates a Maven project, importing the Artemis JMS extension.
+
+== Starting an Artemis server
+
+Then, we need an Artemis server.
+You can follow the instructions from the https://activemq.apache.org/components/artemis/[Apache Artemis web site] or create a `docker-compose.yaml` file with the following content:
+
+[source, yaml]
+----
+# A docker compose file to start an Artemis server
```

Comment:
Is docker-compose really necessary? I managed to run Artemis JMS using just docker:

```docker run -it --rm -p 61616:61616 -e ARTEMIS_USERNAME=myuser -e ARTEMIS_PASSWORD=otherpassword vromero/activemq-artemis```

Reply:
No, it isn't, but it was copied from [AMQP guide](https://github.com/quarkusio/quarkus/blob/master/docs/src/main/asciidoc/amqp-guide.adoc#starting-an-amqp-broker)

Reply:
Also fixed in the corresponding quarkus-quickstarts